src/components/ProductsList/ProductsList.tsx

```diff
@@ -1,20 +1,37 @@
 import ProductCard from '@components/ProductCard/ProductCard';
 import {Product} from '@/store/slices/products/products.model';
 import {List, ListItem} from '@mui/material';
+import {useAppDispatch, useAppSelector} from '@/store/hooks';
+import {selectProducts} from '@/store/slices/products/products.selectors';
+import SecondaryTitle from '../typography/SecondaryTitle/SecondaryTitle';
+import {useEffect} from 'react';
+import {getAllProducts} from '@/store/slices/products/products.thunks';
+import {useSearchParams} from 'react-router-dom';
 
-interface ProductsListProps {
-  products: Product[];
-}
+const ProductsList = () => {
+  const dispatch = useAppDispatch();
+  const products = useAppSelector(selectProducts);
+  const [searchParams] = useSearchParams();
+
+  useEffect(() => {
+    const params = Object.fromEntries(searchParams.entries());
+    dispatch(getAllProducts(params));
+  }, []);
 
-const ProductsList = ({products}: ProductsListProps) => {
   return (
-    <List>
-      {products.map((product: Product) => (
-        <ListItem key={product._id} sx={{mb: 4}}>
-          {<ProductCard product={product} />}
-        </ListItem>
-      ))}
-    </List>
+    <>
+      {products.length === 0 ? (
+        <SecondaryTitle sx={{textAlign: 'center'}} text="Nothing found" />
+      ) : (
+        <List>
+          {products.map((product: Product) => (
+            <ListItem key={product._id} sx={{mb: 4}}>
+              {<ProductCard product={product} />}
+            </ListItem>
+          ))}
+        </List>
+      )}
+    </>
   );
```

Comment:
`<></>` aren't needed here. And it will be probably better vise versa, to have the more important the first
` {Boolean(products.length)  ? (
         <List>
          {products.map((product: Product) => (
            <ListItem key={product._id} sx={{mb: 4}}>
              {<ProductCard product={product} />}
            </ListItem>
          ))}
        </List>
      ) : (
        <SecondaryTitle sx={{textAlign: 'center'}} text="Nothing found" />
      )}`

Reply:
Thanks, changed. But it seems that <></> are needed... I am not sure why.